util/dashboard/gen_dashboard_entry.py

```diff
@@ -33,11 +35,103 @@ def genout(outfile, msg):
     'V3': 'Verification Complete'
 }
 
+# TODO: This is relative to the dashboard, which is currently located at
+# hw/_index.md.
+docs_server = "../.."
+
 
 def convert_stage(stagestr):
     return STAGE_STRINGS.get(stagestr, "UNKNOWN")
 
 
+# Link name with design spec
+def get_linked_design_spec(obj):
+    result = ""
+    if 'design_spec' in obj.keys():
+        url = docs_server + "/" + html.escape(obj['design_spec'])
+        result = "<span title='Design Specification'><a href=\"{}\">{}</a></span>".format(
+            url, html.escape(obj['name'].upper()))
+    else:
+        result = html.escape(obj['name'].upper())
+
+    return result
+
+
+# Link name with dv_plan
+def get_linked_dv_plan(obj):
+    if 'dv_plan' in obj.keys():
+        url = docs_server + "/" + html.escape(obj['dv_plan'])
+        return "<span title='DV Plan'><a href=\"{}\">DV</a></span>".format(url)
+    else:
+        return ""
+
+
+# Link D/V stages with checklist
+def get_linked_checklist(obj, rev, stage, is_latest_rev=True):
+    if not stage or stage not in rev: return ""
+
+    # if in D0 or V0 stage, there is no checklist to link.
+    if rev[stage] in ["D0", "V0"]: return html.escape(rev[stage])
```

Comment:
I would still put a link to the checklist for D0, V0 designs. Being in D0/V0 does not mean that none of the checklist items are fulfilled and it is still useful to see the status. If there no checklist yet (USBDEV, ENTROPY_SRCE), I would put a link to the general checklist to make the table look nicer.

Reply:
> I see. By general checklist, do you mean the template one?

If the life stage is L0, then the D/V # is meaningless (which is true in case of `entropy_src`). For `usbdev` that is at L1 : D0 : V0 and does not have a checklist yet, I have linked the stages to the template at `doc/project/ip_checklist.md.tpl`.

Reply:
SGTM